go/api/glide_cluster_client.go

```diff
@@ -311,3 +315,147 @@ func (client *GlideClusterClient) EchoWithOptions(echoOptions options.ClusterEch
+
+	if payload.error != nil {
+		return nil, payload.error
+	}
+
+	return payload.value, nil
+}
+
+// Incrementally iterates over the keys in the cluster.
+// The method returns a list containing the next cursor and a list of keys.
+//
+// This command is similar to the SCAN command but is designed to work in a cluster environment.
+// For each iteration, a new cursor object should be used to continue the scan.
+// Using the same cursor object for multiple iterations will result in the same keys or unexpected behavior.
+// For more information about the Cluster Scan implementation, see
+// https://github.com/valkey-io/valkey-glide/wiki/General-Concepts#cluster-scan.
+//
+// Like the SCAN command, the method can be used to iterate over the keys in the database,
+// returning all keys the database has from when the scan started until the scan ends.
+// The same key can be returned in multiple scan iterations.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	cursor - The [ClusterScanCursor] object that wraps the scan state.
+//	   To start a new scan, create a new empty ClusterScanCursor using NewClusterScanCursor().
+//
+// Returns:
+//
+//	The ID of the next cursor.
+//	List of keys found for this cursor ID.
+//
+// [valkey.io]: https://valkey.io/commands/scan/
+func (client *GlideClusterClient) Scan(
+	cursor *options.ClusterScanCursor,
+) (string, []string, error) {
+	response, err := client.clusterScan(cursor, nil)
+	if err != nil {
+		fmt.Println(err)
+		return DefaultStringResponse, []string{}, err
+	}
+
+	return handleScanResponse(response)
+}
+
+// Incrementally iterates over the keys in the cluster.
+// The method returns a list containing the next cursor and a list of keys.
+//
+// This command is similar to the SCAN command but is designed to work in a cluster environment.
+// For each iteration, a new cursor object should be used to continue the scan.
+// Using the same cursor object for multiple iterations will result in the same keys or unexpected behavior.
+// For more information about the Cluster Scan implementation, see
+// https://github.com/valkey-io/valkey-glide/wiki/General-Concepts#cluster-scan.
+//
+// Like the SCAN command, the method can be used to iterate over the keys in the database,
+// returning all keys the database has from when the scan started until the scan ends.
+// The same key can be returned in multiple scan iterations.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	cursor - The [ClusterScanCursor] object that wraps the scan state.
+//	   To start a new scan, create a new empty ClusterScanCursor using NewClusterScanCursor().
+//	opts - The scan options. Can specify MATCH, COUNT, and TYPE configurations.
+//
+// Returns:
+//
+//	The ID of the next cursor.
+//	List of keys found for this cursor ID.
+//
+// [valkey.io]: https://valkey.io/commands/scan/
+func (client *GlideClusterClient) ScanWithOptions(
+	cursor *options.ClusterScanCursor,
+	opts *options.ClusterScanOptions,
```

Comment:
```suggestion
	opts options.ClusterScanOptions,
```
So a user can't pass `nil` instead of the pointer